staging/src/k8s.io/sample-controller/controller.go

```diff
@@ -154,10 +154,10 @@ func (c *Controller) Run(workers int, stopCh <-chan struct{}) error {
 
 	// Start the informer factories to begin populating the informer caches
 	klog.Info("Starting Foo controller")
-
+	defer klog.Info("Shutting down Foo controller")
 	// Wait for the caches to be synced before starting workers
 	klog.Info("Waiting for informer caches to sync")
-	if ok := cache.WaitForCacheSync(stopCh, c.deploymentsSynced, c.foosSynced); !ok {
+	if ok := cache.WaitForNamedCacheSync("Foo", stopCh, c.deploymentsSynced, c.foosSynced); !ok {
```

Comment:
I'm not sure what the optimization here is?
`WaitForNamedCacheSync` additionally just emits logs:
https://github.com/kubernetes/client-go/blob/7f0455096073f978a0f4b832c7ab10a0ed5a66b9/tools/cache/shared_informer.go#L254-L264